pkg/virt-operator/resource/apply/certificates.go

```diff
@@ -7,25 +7,43 @@ import (
 )
 
 func getCADuration(config *k8sv1.KubeVirtSelfSignConfiguration) *metav1.Duration {
-	if config == nil || config.CARotateInterval == nil {
-		return &metav1.Duration{Duration: Duration7d}
+	if config != nil {
+		// deprecated, but takes priority to provide a smooth upgrade path
+		if config.CARotateInterval != nil {
```

Comment:
Should a validating webhook ensure that the old and new APIs are not used at the same time (i.e., are mutually exclusive)?

Reply:
I'm unsure of the value in this. The point of leaving the old definitions in at all is to ease upgrade burden. Enforcing use of one or the other might increase the potential to introduce upgrade complexity.

As it stands, one completely overrides the other so I'm not sure adding a validating webhook is of much value. What do you think?

Reply:
I think the value is in preventing a contradicting configuration between old and new which is silently ignored. With such validation, the contradicting configuration is early discovered (at "compile time" :-D) and can be fixed.

Reply:
Upon further investigation, virt-api does not have a validating webhook for KubeVirt objects.

This makes sense. Though this codebase defines the object, it is not the owner of any objects created, so it should not be asserting control over the contents. Ensuring new values are set is HCO's responsibility.